oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,169 @@
+
+The settings found in both the ``required.py`` and ``defaults.py`` files will be documented to describe what they are and how they should be used.  The documentation will consist of `Sphinx autodoc`_ compatible comments before each setting.  For reference that is a ``“comment with special formatting (using a #: to start the comment instead of just #)”``.  This keeps documentation close to the code as it is being written, while allowing it to be surfaced in generated docs.
+ 
+
+.. _Sphinx autodoc: https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoattribute
+
+Operations Manuals
+******************
+
+A clear manual of operations will exist in the form of RST files in an ``operations`` directory within the ``documentation`` directory for that application.  See `this commit`_ for an example provided by the Open edX Build-Test-Release working group. The operations docs will cover common operations such as how to run the application for web traffic or as an async worker and how to manage the applications underlying database schema.  It will also include a list of potential maintenance tasks operators may want to leverage such as clearing sessions or applying security patches.  Finally it will include the list of ad-hoc management commands operators can use to help handle edge case or one-time operations.
+ 
+In the same vein as not dictating how operators create and manage their application config files, so too will operators be expected to manage how they execute the operations documented in the manual.
+
+.. _this commit: https://github.com/openedx-btr-wg/edx-platform/commit/        18effd83f983f497ca0a1535108fa41dc50d06a2#diff-ca02329742db0a77612a18ba1260d178R1-R39
+
+
+Alternatives Considered
+=======================
+
+Refactor configuration
+**********************
+
+Refactoring the configuration repo to reduce the amount of shared code and making it easier to read and understand what is being run when you execute a particular playbook.
+
+This approach was rejected due to the sheer volume of work required to make this change in an environment that is inherently difficult to test.  Also because while it would improve ease of use, it would only provide parity in terms of functionality.  Moving to containers will also improve ease of use as well as unlocking many potential future enhancements.
+
+Jumping to Kubernetes
+*********************
+
+Kubernetes is an open source container orchestration platform pioneered by Google.  While it often occupies the same conversation space as containers because it is a powerful way to manage them, it is a huge increase in complexity and expertise required to operate.  For most installations Kubernetes is currently too much overhead/learning curve for the value.  The edX organization may opt to explore deploying Docker containers this way in the future and would love to collaborate with operators who also decide to use Kubernetes to compare notes.
+
```

Comment:
Could we consider using environment variables for configuration, even if it's just to dismiss them? IMHO there are simply to many complex (dicts and lists) variables to be defined in edx-platform to set them all as environment variables.